packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts

```diff
@@ -420,7 +420,11 @@ export function frameSnapshotStreamer(snapshotStreamer: string) {
             if (nodeName === 'IFRAME' && name === 'src')
               continue;
             let value = element.attributes[i].value;
-            if (name === 'src' && (nodeName === 'IMG'))
+            if (name === 'charset' && nodeName === 'META')
+              value = 'utf-8';
+            else if (nodeName === 'META' && (node as HTMLMetaElement).httpEquiv.toLowerCase() === 'content-type' && name === 'content')
+              value = 'text/html; charset=utf-8';
```

Comment:
Let's only rewrite the charset inside the content? I'm worried about non-html frames, for example `<meta http-equiv="Content-Type" content="image/svg+xml">`.

I'd also make a separate function like `value = this._sanitizeMetaAttribute(name, value, (node as HTMLMetaElement).httpEquiv)`

Reply:
I implemented `_sanitizeMetaAttribute`.

Reply:
but `_sanitizeMetaAttribute` may not be needed because `meta` tag is html specific.